### ISSUE TITLE
Inverse Quadratic derivative method 

### ISSUE THREAD

**Issue by jherekhealy** [JuliaMath/Roots.jl#226]
Halley's method does not always behave best further away from the zero.
An old but robust alternative is the inverse quadratic method. 
Its implementation would look like

```julia
struct InverseQuadraticMethod <: Roots.AbstractHalleyLikeMethod
end

function Roots.update_state(method::InverseQuadraticMethod, fs, o::Roots.UnivariateZeroState{T,S}, options::Roots.UnivariateZeroOptions) where {T,S}
    xn = o.xn1
    fxn = o.fxn1
    r1, r2 = o.m

    xn1::T = xn - (1+ r1/(r2*2))*r1 

    tmp = Roots.fΔxΔΔx(fs, xn1)
    fxn1::S, r1::T, r2::T = tmp[1], tmp[2], tmp[3]
    Roots.incfn(o,3)

    o.xn0, o.xn1 = xn, xn1
    o.fxn0, o.fxn1 = fxn, fxn1
    empty!(o.m); append!(o.m, (r1, r2))
end
```

I will create a pull request if interested.

**Comment by jverzani** [comment]
I not so long ago added the family `Roots.LithBoonkkampIJzerman` which likely covers this, but, admittedly, that method is more interesting to me as I like parameterized families. So yes, a PR with a simpler, more discoverable method would be most welcome.

**Comment by jverzani** [comment]
Closed by #248
